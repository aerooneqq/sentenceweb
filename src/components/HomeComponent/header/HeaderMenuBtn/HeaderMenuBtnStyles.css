@media screen and (min-width: 560px){ 
    #headerMenuBtnCont{ 
        display: none;
    }
}

@media screen and (max-width: 560px){ 
    #headerMenuBtnCont{ 
        width: 50px;
        height: 50px; 
    
        display: flex;
        flex-direction: column;
        align-items: center; 
        justify-content: center;

        margin-right: 15px;
    
        background: transparent;
    }
}

.headerMenuBtnLine{ 
    width: 40px;
    height: 1px;

    background: white;
}

.headerMenuBtnLine:nth-child(2){ 
    margin-top: 10px;
}

.headerMenuBtnLineOpened:nth-child(1){
    animation: topLineRotation 0.1s ease-in forwards, 
               moveLineDown 0.1s ease-in forwards; 
}

    @keyframes topLineRotation{ 
        from{ 
            transform: rotateZ(0deg);
        }

        to{ 
            transform: rotateZ(45deg);
            transform-origin: center;
        }
    }

    @keyframes moveLineDown{
        from{ 
            margin-top: 0px;
        }

        to{ 
            margin-top: 1px;
        }
    }

.headerMenuBtnLineOpened:nth-child(2){
    animation: bottomLineRotation 0.1s ease-in forwards, 
               moveLineUp 0.1s ease-in forwards; 
}


    @keyframes bottomLineRotation{ 
        from{ 
            transform: rotateZ(0deg);
        }

        to{ 
            transform: rotateZ(-45deg);
            transform-origin: center;
        }
    }

    @keyframes moveLineUp{ 
        from{ 
            margin-top: 0px;
        }

        to{ 
            margin-top: -1px;
        }
    }

.headerMenuBtnLineClosed:nth-child(1){ 
    animation: topLineRotationReverse 0.1s ease-in forwards, 
               moveLineDownReverse 0.1s ease-in forwards; 
}

    @keyframes topLineRotationReverse{ 
        from{ 
            transform: rotateZ(45deg);
            transform-origin: center;
        }

        to{ 
            transform: rotateZ(0deg);
        }
    }

    @keyframes moveLineDownReverse{
        from{ 
            margin-top: 1px;
        }

        to{ 
            margin-top: 0px;
        }
    }

.headerMenuBtnLineClosed:nth-child(2){ 
    animation: bottomLineRotationReverse 0.1s ease-in forwards, 
               moveLineUpReverse 0.1s ease-in forwards; 
}

    @keyframes bottomLineRotationReverse{ 
        from{ 
            transform: rotateZ(-45deg);
            transform-origin: center;
        }

        to{ 
            transform: rotate(0deg);
        }
    }

    @keyframes moveLineUpReverse{ 
        from{ 
            margin-top: -1px;
        }

        to{ 
            margin-top: 10px;
        }
    }