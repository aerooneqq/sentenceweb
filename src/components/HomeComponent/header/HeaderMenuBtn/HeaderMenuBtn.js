import React, {Component} from "react"

//Styles
import "./HeaderMenuBtnStyles.css"

export default class HeaderMenuBtn extends Component{
    constructor(props){ 
        super(props);

        this.closed = true;
        this.handleMenuBtnClick = this.handleMenuBtnClick.bind(this);
        this._rotate = this._rotate.bind(this);
        this._changeStateWhenClicked = this._changeStateWhenClicked.bind(this);
    }

    _rotate() { 
        if (this.closed) { 
            document.getElementById("headerMenuBtnTopLine").classList.remove("headerMenuBtnLineClosed");
            document.getElementById("headerMenuBtnTopLine").classList.toggle("headerMenuBtnLineOpened");
            document.getElementById("headerMenuBtnBottomLine").classList.remove("headerMenuBtnLineClosed");
            document.getElementById("headerMenuBtnBottomLine").classList.toggle("headerMenuBtnLineOpened");
        }
        else { 
            document.getElementById("headerMenuBtnTopLine").classList.remove("headerMenuBtnLineOpened");
            document.getElementById("headerMenuBtnTopLine").classList.toggle("headerMenuBtnLineClosed");
            document.getElementById("headerMenuBtnBottomLine").classList.remove("headerMenuBtnLineOpened");
            document.getElementById("headerMenuBtnBottomLine").classList.toggle("headerMenuBtnLineClosed");
        }
    }

    _changeStateWhenClicked() { 
        if (this.closed){ 
            this.props.showOrHideMenuItems(this.closed);
            this.closed = false;
        }
        else { 
            this.props.showOrHideMenuItems(this.closed);
            this.closed = true;
        }
    }

    handleMenuBtnClick(){
        this._changeStateWhenClicked();
        this._rotate();
    }

    render() { 
        return ( 
            <div id = "headerMenuBtnCont" onClick = {this.handleMenuBtnClick}>
                <div id = "headerMenuBtnTopLine" className = "headerMenuBtnLine"  />
                <div id = "headerMenuBtnBottomLine" className = "headerMenuBtnLine" />
            </div>
        );
    }
}