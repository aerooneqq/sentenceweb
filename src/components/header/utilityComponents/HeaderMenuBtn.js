import React, {Component} from "react"

//Styles
import "../styles/HeaderMenuBtnStyles.css"

export default class HeaderMenuBtn extends Component{ 
    constructor(props){ 
        super(props);
    }

    render() { 
        return ( 
            <div id = "headerMenuBtnCont">
                <div className = "headerMenuBtnLine" />
                <div className = "headerMenuBtnLine" />
            </div>
        );
    }
}