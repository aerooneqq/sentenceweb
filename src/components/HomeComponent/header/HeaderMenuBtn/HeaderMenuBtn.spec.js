import React from 'react'
import { shallow } from 'enzyme'
import {shallowToJson} from "enzyme-to-json"
import HeaderMenuBtn from './HeaderMenuBtn';

describe("Header menu btn", () => { 
    it("should render correctly", () => { 
        const headerMenuBtn = shallow(<HeaderMenuBtn />);

        expect(shallowToJson(headerMenuBtn)).toMatchSnapshot();
    });
});