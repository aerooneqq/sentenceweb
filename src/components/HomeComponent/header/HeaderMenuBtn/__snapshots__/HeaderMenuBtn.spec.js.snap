// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header menu btn should render correctly 1`] = `
<div
  id="headerMenuBtnCont"
  onClick={[Function]}
>
  <div
    className="headerMenuBtnLine"
    id="headerMenuBtnTopLine"
  />
  <div
    className="headerMenuBtnLine"
    id="headerMenuBtnBottomLine"
  />
</div>
`;
